### PR TITLE
fix(formation): enrich item owner and activity actor display names

### PR DESCRIPTION
**Pull request by andrest50**
## Summary

- Adds `name: string | null` to `FormationPersonMetadata` — derived from `given_name`/`family_name` (or the top-level `name` field as fallback) via the existing `USER_METADATA_READ` NATS call
- Adds `resolveDisplayName()` static helper to `FormationService` to pick the best available display name from a raw metadata profile
- Adds `enrichFormationUserRefs()` private method that batch-resolves display names for a set of `FormationUser` refs (owners, actors) using the existing `readUserMetadata()` process-wide cache and `settleInBatches()` budget — no new NATS round-trips when the same username was already fetched for the People panel
- Wires `enrichFormationUserRefs()` into `getProjectFormation` (item owners) and `fetchItemActivityOrDegrade` (activity actors), replacing the username-as-name placeholder the mapper sets
- Updates `formation.service.spec.ts` to include `name` in the `userMetadataCacheValueForTests` expectation

## Test Plan

- [ ] `yarn check-types` passes
- [ ] `yarn lint` passes
- [ ] Load a checklist with an assigned item: `owner.name` should show the display name ("Andres Tobon") not the username ("andrest50")
- [ ] Open an item's activity drawer: actor names should show display names

## Related

- Backend display-name fix: https://github.com/linuxfoundation/lfx-v2-formation-service/pull/23

## Ticket

[lfx-self-serve-ops#2576](https://github.com/linuxfoundation/lfx-self-serve-ops/issues/2576)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/lfx-one/src/server/services/formation.service.ts [hunk 1]
```diff
@@ -1459,7 +1512,15 @@ export class FormationService {
...
         itemUid
       );
-      return { history: entries, history_state: 'complete' };
+
+      // Enrich actor display names — mapActor sets name = username as a placeholder for real users.
+      const actorNames = await this.enrichFormationUserRefs(
+        req,
+        entries.map((e) => e.actor)
+      );
+      const enrichedEntries = actorNames.size > 0 ? entries.map((e) => (actorNames.has(e.actor.username) ? { ...e, actor: { ...e.actor, name: actorNames.get(e.actor.username)! } } : e)) : entries;
```
Comment (dealako) [inline apps/lfx-one/src/server/services/formation.service.ts]: [blocking] CI red: `yarn format:check` fails on this line

Issue: This line is 196 characters, over the repo's prettier `printWidth`. The `Code Quality Checks` job on this head SHA fails at `yarn format:check`, so `yarn lint:check`/`yarn check-types` never even ran for this commit.
Proof: `awk 'length>180' formation.service.ts` flags line 1521; CI run for b157eafb fails with `Process completed with exit code 1` at the format step.
Why it matters: the PR cannot merge as-is, and any lint/type regressions in this diff are currently unverified by CI.
Fix: run `yarn format` (prettier will wrap the ternary) and re-push. While you're in here, this line also nests a ternary inside a `.map()` callback — CLAUDE.md says "Never nest ternary expressions." The owner-merge a few hundred lines up (~line 215) uses a plain `for` loop for the equivalent case; mirroring that style here would fix both the length and the nesting at once.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/lfx-one/src/server/services/formation.service.ts [hunk 2]
```diff
@@ -1155,6 +1207,7 @@ export class FormationService {
     }
 
     return {
+      name: FormationService.resolveDisplayName(profile),
```
Comment (dealako) [inline apps/lfx-one/src/server/services/formation.service.ts:1242]: [blocking] Likely `yarn check-types` failure: `UserMetadata` interface isn't assignable to `Record<string, unknown>`

Issue: `profile` here is typed `UserMetadata` (an `interface`, from `packages/shared/src/interfaces/user-profile.interface.ts`), but `resolveDisplayName` (line 1099) declares its parameter as `Record<string, unknown>`. TypeScript does not grant an implicit index signature to a plain interface, so passing an interface-typed value where an indexed type is expected is a compile error (`TS2345: Index signature for type 'string' is missing in type 'UserMetadata'`).
Proof: reproduced standalone with `tsc --strict --noEmit` against the same shape (interface with optional string props → function expecting `Record<string, unknown>`) — it fails with exactly that TS2345. This wasn't caught by CI here because the format-check failure (see the other comment on this PR) aborted the job before `check-types` ran.
Why it matters: the branch doesn't compile.
Fix: type the parameter as `profile: UserMetadata` instead of `Record<string, unknown>` — dot access still works with the existing `typeof x === 'string'` guards (e.g. `typeof profile.given_name === 'string'`), and it keeps this in line with the sibling projection code a few lines below, which already uses typed dot access (`profile.job_title`, `profile.organization`, `profile.picture`) rather than bracket indexing. Verify with `yarn check-types` after fixing.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/lfx-one/src/server/services/formation.service.ts [hunk 3]
```diff
@@ -1078,6 +1091,45 @@ export class FormationService {
...
+    const targets = [...new Set(refs.filter((r) => r.username && r.username !== 'system').map((r) => r.username))];
+    if (targets.length === 0) return new Map();
+
+    const results = await settleInBatches(targets, FORMATION_PEOPLE_ENRICHMENT_BATCH_SIZE, (username) => this.readUserMetadata(req, username), {
+      deadlineAt: Date.now() + FORMATION_PEOPLE_ENRICHMENT_BUDGET_MS,
+    });
+
+    const nameByUsername = new Map<string, string>();
+    results.forEach((result, index) => {
+      if (result.status === 'fulfilled' && result.value?.name) {
```
Comment (dealako) [inline apps/lfx-one/src/server/services/formation.service.ts:1141]: [minor] `enrichFormationUserRefs` silently swallows lookup failures and budget exhaustion

Issue: non-fulfilled results (`rejected`, or a skip from `BatchDeadlineExceededError`) are dropped with no logging at all here. Its sibling `enrichFormationPeople` (~line 1051-1080) distinguishes rejected vs. budget-exceeded vs. fulfilled and logs each (masked username on failure, one aggregate warning on budget exhaustion).
Proof: this `results.forEach` only handles the `status === 'fulfilled' && result.value?.name` case; nothing else is inspected.
Why it matters: when names silently stay as raw usernames in production, there's no log signal to distinguish "profile has no name" from "auth-service is down" or "budget exhausted" — the degradation path is undiagnosable.
Fix: mirror `enrichFormationPeople`'s result loop — count `BatchDeadlineExceededError` skips into one aggregate warning, and `logger.warning` (with `maskIdentifierForLogs(username)`) on other rejections.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/lfx-one/src/server/services/formation.service.ts [hunk 4]
```diff
@@ -1078,6 +1091,45 @@ export class FormationService {
...
+  /**
+   * Batch-enriches an array of {@link FormationUser} refs — the item `owner` and activity `actor`
+   * shapes — by replacing the username-as-name placeholder with the actual display name from the
+   * auth-service user-metadata read. Reuses {@link readUserMetadata}'s process-wide cache, so names
+   * fetched during a concurrent {@link enrichFormationPeople} call are cache hits here. System actors
+   * (username `"system"`) are skipped; their name is already set to `"System"` by the mapper. Best-
+   * effort: a failed or budget-exceeded lookup leaves the ref with its username-as-name placeholder.
+   */
+  private async enrichFormationUserRefs(req: Request, refs: FormationUser[]): Promise<Map<string, string>> {
+    const targets = [...new Set(refs.filter((r) => r.username && r.username !== 'system').map((r) => r.username))];
```
Comment (dealako) [inline apps/lfx-one/src/server/services/formation.service.ts]: [nit] `'system'` sentinel duplicated as a bare string literal

Issue: this skip relies on the literal `'system'`, duplicating `mapActor`'s own literal in `formation-activity.helper.ts` (`raw.set_by === 'system' || raw.actor === 'system'`), with nothing tying the two together.
Proof: no shared constant exists for this in `packages/shared/src/constants/`.
Why it matters: if the sentinel ever changes, this skip silently stops working and `system` becomes a live (and pointless) auth-service lookup on every activity read.
Fix: export a `FORMATION_SYSTEM_ACTOR_USERNAME` constant from `@lfx-one/shared/constants` and use it in both places.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by dealako — CHANGES_REQUESTED** [review]
Solid, well-scoped fix for the underlying issue, but the branch currently fails CI (format check, and very likely a type-check error hiding behind it) — see inline comments for both, plus a handful of minor/nit items.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by dealako** [comment]
Hey @andrest50, thanks for the quick turnaround on this — the username-as-display-name issue has been a visible papercut in the checklist and activity drawer for a while.

**Overall impression:** this is a clean, well-scoped fix. The enrichment approach (reusing the existing process-wide metadata cache and `settleInBatches` budget instead of adding new NATS round-trips) is the right call, and the privacy angle is fine — the `name` field is projected the same way the existing `job_title`/`organization`/`picture` fields already are, and it doesn't leak to any less-privileged audience than already sees the same name via the People panel. The main blocker is that the branch doesn't currently pass CI.

🔴 **Blocking: 2 issues:** `yarn format:check` fails on a 196-char line (formation.service.ts:1521), and that failure is masking what looks like a real `yarn check-types` error — `resolveDisplayName`'s `profile: Record<string, unknown>` parameter isn't assignable from the `UserMetadata` interface type actually passed in at the call site (confirmed by reproducing the error standalone with `tsc --strict`).

🟡 **Minor: 6 issues:** `enrichFormationUserRefs` swallows lookup failures/budget exhaustion with no logging (unlike its sibling `enrichFormationPeople`), owner-name enrichment is only wired into the checklist list read and not into the item-detail/write paths (so the same item can show a resolved name in the row and a raw username in the drawer), the process-wide cache's doc comment is now stale (still says "title, organization, picture" and "people card" only). Copilot also caught three real ones I agree with and won't duplicate: the given/family-name fallback logic drops the `name` fallback when only one part is present, owner metadata lookup adds unnecessary serial latency to the checklist response, and the new logic (resolveDisplayName precedence, batching/dedup, system-actor skip) has no direct test coverage.

⚪ **Nit: 2 issues:** a duplicated `'system'` string literal between this file and the activity mapper with no shared constant, and a doc comment on `readUserMetadata` that undercounts its callers after this PR.

🔴 **Needs changes before approval**

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/lfx-one/src/server/services/formation.service.ts [hunk 5]
```diff
@@ -209,6 +210,18 @@ export class FormationService {
       mapUpstreamFormationItem(raw, { formationUid: `formation:${uid}`, projectUid: uid, projectSlug: project.slug, sectionTitles })
     );
 
+    // Enrich item owner display names — the mapper sets name = username as a placeholder; replace
+    // with the actual profile name where available. Best-effort; items without a name hit stay as-is.
+    const ownerRefs = items.flatMap((item) => (item.owner ? [item.owner] : []));
+    const ownerNames = await this.enrichFormationUserRefs(req, ownerRefs);
```
[the remark this answers is not in the document]
Comment (andrest50) [reply apps/lfx-one/src/server/services/formation.service.ts]: Fixed — unique assignees are now extracted from the raw checklist (already fetched) before the `Promise.all([project, team, root, settings])` block, so NATS lookups run concurrently with those four reads. The resolved map is `await`ed only after item mapping (commit e5db828).

File: apps/lfx-one/src/server/services/formation.service.ts [hunk 6]
```diff
@@ -1078,6 +1091,45 @@ export class FormationService {
...
+   * `given_name + family_name` when both are present and non-blank; falls back to the top-level
+   * `name` field; returns `null` when nothing usable is available.
+   */
+  private static resolveDisplayName(profile: Record<string, unknown>): string | null {
+    const given = typeof profile['given_name'] === 'string' ? profile['given_name'].trim() : '';
+    const family = typeof profile['family_name'] === 'string' ? profile['family_name'].trim() : '';
+    if (given || family) {
+      return [given, family].filter(Boolean).join(' ') || null;
+    }
+    return FormationService.metadataText(profile['name']);
```
[the remark this answers is not in the document]
Comment (andrest50) [reply apps/lfx-one/src/server/services/formation.service.ts]: Fixed — changed the guard to `if (given && family)` so both parts must be non-blank before joining. The fallback chain is now: given+family → top-level `name` → lone part → `null`. That avoids skipping `name` when only one part is present (commit e5db828).

File: apps/lfx-one/src/server/services/formation.service.spec.ts [hunk 7]
```diff
@@ -378,6 +378,7 @@ describe('FormationService', () => {
 
       expect(result.people[0]).toEqual(expect.objectContaining({ job_title: 'Partner contact', organization: null, avatar: 'https://cdn.example/s.png' }));
       await expect(FormationService.userMetadataCacheValueForTests('sam.chen')).resolves.toEqual({
+        name: null,
```
[the remark this answers is not in the document]
Comment (andrest50) [reply apps/lfx-one/src/server/services/formation.service.spec.ts:381]: Fixed — added four tests covering: owner name enriched via `given_name`/`family_name` (e2e through NATS mock), owner falls back to username when no metadata, actor name enriched from metadata, system actor returns `{ username: 'system', name: 'System' }` with no NATS call issued. The display-name precedence test (given+family beats top-level name, lone part last resort) also runs through `getFormationPeople` and checks the cache value (commit e5db828).

File: apps/lfx-one/src/server/services/formation.service.ts (hunk 1 above)
Comment (andrest50) [reply apps/lfx-one/src/server/services/formation.service.ts]: Fixed — replaced the 196-char nested-ternary `.map()` with a plain `for` loop matching the owner-enrichment pattern. `yarn format:check` passes (commit e5db828).

File: apps/lfx-one/src/server/services/formation.service.ts (hunk 2 above)
Comment (andrest50) [reply apps/lfx-one/src/server/services/formation.service.ts:1242]: Fixed — `resolveDisplayName` now takes `UserMetadata` and uses dot-access (`profile.given_name` etc.) instead of bracket access on `Record<string, unknown>`. `yarn check-types` passes cleanly (commit e5db828).

File: apps/lfx-one/src/server/services/formation.service.ts (hunk 3 above)
Comment (andrest50) [reply apps/lfx-one/src/server/services/formation.service.ts:1141]: Fixed — `enrichFormationUserRefs` now emits a `logger.warning` per rejected lookup (with masked username and `err`) and a single aggregate warning when budget is exhausted (with `budget_ms` and `skipped` count). Fulfilled results with no resolvable name are still silent (expected degradation) (commit e5db828).

File: apps/lfx-one/src/server/services/formation.service.ts (hunk 4 above)
Comment (andrest50) [reply apps/lfx-one/src/server/services/formation.service.ts]: Fixed — exported `FORMATION_SYSTEM_ACTOR_USERNAME = 'system'` from `formation-people.constants.ts`. Both `formation.service.ts` (`enrichFormationUserRefs` skip) and `formation-activity.helper.ts` (`mapActor`) import and use the constant (commit e5db828).

File: apps/lfx-one/src/server/services/formation.service.spec.ts [hunk 8]
```diff
@@ -790,6 +791,48 @@ describe('FormationService', () => {
...
+      // u1: both parts — name should be "Ada Lovelace", NOT "Ada Lovelace Full" (given+family wins)
+      // u2: only given_name — falls through to top-level name "Full Name"
+      natsRequest.mockImplementation(async (_subject: string, username: string) => {
+        if (username === 'u1')
+          return { data: JSON.stringify({ success: true, data: { given_name: 'Ada', family_name: 'Lovelace', name: 'Ada Lovelace Full' } }) };
+        if (username === 'u2') return { data: JSON.stringify({ success: true, data: { given_name: 'Ada', name: 'Full Name' } }) };
+        return { data: '' };
+      });
+
+      await service.getFormationPeople(buildReq(), 'live-project');
```
Comment (dealako) [inline apps/lfx-one/src/server/services/formation.service.spec.ts:839]: [blocking] New precedence test fails — `yarn test` is red on this head

Issue: this test never stubs `proxyRequest`, unlike every sibling test in the file (e.g. lines 796, 808). `getFormationPeople` first calls `fetchLiveChecklistOrDenyNotFound`, which reads the (unmocked) checklist and passes it to `sectionTitlesFromChecklist`, which unconditionally dereferences `raw.sections`.
Proof: `cd apps/lfx-one && yarn vitest run src/server/services/formation.service.spec.ts` → `Tests 1 failed | 136 passed`.
```
FAIL  ... > resolveDisplayName: given+family beats top-level name; lone part is a last resort
TypeError: Cannot read properties of undefined (reading 'sections')
 ❯ sectionTitlesFromChecklist src/server/helpers/formation-mapper.helper.ts:172:22
 ❯ FormationService.fetchLiveChecklistOrDenyNotFound src/server/services/formation.service.ts:1304:33
 ❯ FormationService.getFormationPeople src/server/services/formation.service.ts:280:5
 ❯ src/server/services/formation.service.spec.ts:831:7
```
Why it matters: the CI "Run unit tests" step (`yarn test`) will fail on this commit, and the precedence assertions this test exists to lock in are never reached.
Fix: add `proxyRequest.mockResolvedValue(checklist([rawItem()]));` before the `getFormationPeople` call, matching the pattern used at lines 796/808.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/lfx-one/src/server/services/formation.service.ts [hunk 9]
```diff
@@ -1078,6 +1101,67 @@ export class FormationService {
...
+   */
+  private static resolveDisplayName(profile: UserMetadata): string | null {
+    const given = typeof profile.given_name === 'string' ? profile.given_name.trim() : '';
+    const family = typeof profile.family_name === 'string' ? profile.family_name.trim() : '';
+    if (given && family) {
+      return `${given} ${family}`;
+    }
+    const topName = FormationService.metadataText(profile.name);
+    if (topName) return topName;
+    return given || family || null;
```
Comment (dealako) [inline apps/lfx-one/src/server/services/formation.service.ts:1119]: [minor] The exact edge case this fix targeted — a lone given/family part with no top-level `name` — is still untested

Issue: the new precedence test (once its checklist mock is fixed) only covers given+family (`u1`) and given-only-with-a-top-level-`name` fallback (`u2`, which returns `name`, not this line). No fixture reaches `return given || family || null` with exactly one part present and no top-level `name`.
Proof: `formation.service.spec.ts` lines 822-828 — neither mocked user omits `name` while supplying only one of `given_name`/`family_name`.
Why it matters: this is precisely the input the prior review round flagged as broken (a lone part being dropped instead of used). A future refactor could regress this branch back to `null` and the suite would stay green.
Fix: add a third mocked user (e.g. `family_name` only, no `given_name`/`name`) and assert the cached value resolves to that lone part.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/lfx-one/src/server/services/formation.service.ts [hunk 10]
```diff
@@ -159,7 +162,14 @@ export class FormationService {
     // independent of the others, so they run concurrently rather than as three sequential round
     // trips: the project read, the ROOT-collapse lookup, and the settings read for announcement_date
     // (which degrades to null on its own failure — see its .catch() below — independently of the
-    // other two).
+    // other two). Owner-name enrichment is also kicked off here — it only needs the distinct
+    // assignees from the already-fetched checklist, so its NATS calls are in-flight while the other
+    // reads run rather than serializing after them.
+    const rawAssignees = [...new Set(checklist.items.map((r) => r.assignee).filter((a): a is string => !!a))];
+    const ownerEnrichmentPromise = this.enrichFormationUserRefs(
```
Comment (dealako) [inline apps/lfx-one/src/server/services/formation.service.ts:169]: [nit] `ownerEnrichmentPromise` has no rejection handler attached until it's awaited ~60 lines later

Issue: this promise is created here but only `await`ed at line 226, after the `Promise.all(...)` a few lines down. If that `Promise.all` rejects first, `getProjectFormation` throws without this promise ever being awaited or handled.
Proof: today `enrichFormationUserRefs` can't realistically reject (`settleInBatches` never throws, and the code before its first `await` is a pure filter/`Set`), so this is latent rather than live — but Node's default `unhandledRejection` behavior terminates the process, so any future change that lets it throw before the await turns this into a process crash on an already-failing request path.
Why it matters: latent crash risk on an error path, and in-flight NATS lookups keep running for a request that's already failed.
Fix: attach a defensive handler at creation, e.g. `.catch(() => new Map<string, string>())`, so it's pre-handled regardless of when the `Promise.all` settles.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by dealako — CHANGES_REQUESTED** [review]
Great progress on the prior round's findings — the concurrency, precedence-logic, type, format, and logging fixes all check out. One new test regression needs a fix before this is mergeable; see inline comments.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by dealako** [comment]
@andrest50 thanks for turning this around quickly — this is a solid follow-up. Six of the seven items from the prior round are cleanly fixed, verified against actual `tsc`/`prettier`/`eslint` output rather than just the reply comments. The one open item is that the new test coverage itself introduced a test-suite regression, which is why this round is still a changes-requested.

👏 **Nice work** on: kicking off the owner-metadata NATS lookup concurrently with the `Promise.all` instead of serializing it (`e5db828`), correcting `resolveDisplayName`'s precedence so a lone `given_name`/`family_name` no longer suppresses the top-level `name` fallback, and extracting `FORMATION_SYSTEM_ACTOR_USERNAME` so the `'system'` sentinel isn't duplicated between the service and `formation-activity.helper.ts`.

**Revision tracking (prior round → this round):**
- Owner metadata lookup latency → ✅ resolved — concurrent kickoff before `Promise.all` ([`e5db828`](https://github.com/linuxfoundation/lfx-self-serve/commit/e5db82869988eda5b312146153481a0ce8c73050))
- Name-fallback logic (lone part dropped) → ✅ resolved in the implementation ([`e5db828`](https://github.com/linuxfoundation/lfx-self-serve/commit/e5db82869988eda5b312146153481a0ce8c73050)); ⚠️ the exact lone-part-no-`name` case the fix targeted is still untested (see inline)
- Format/CI-red (196-char line, nested ternary) → ✅ resolved — `yarn format:check` and `yarn lint:check` both verified clean ([`e5db828`](https://github.com/linuxfoundation/lfx-self-serve/commit/e5db82869988eda5b312146153481a0ce8c73050))
- Type error (`Record<string, unknown>` vs `UserMetadata`) → ✅ resolved — verified with a direct `tsc -p tsconfig.app.json --noEmit` ([`e5db828`](https://github.com/linuxfoundation/lfx-self-serve/commit/e5db82869988eda5b312146153481a0ce8c73050))
- Missing logging in `enrichFormationUserRefs` → ✅ resolved — now mirrors `enrichFormationPeople`'s rejected/budget-exceeded logging ([`e5db828`](https://github.com/linuxfoundation/lfx-self-serve/commit/e5db82869988eda5b312146153481a0ce8c73050))
- `'system'` sentinel duplication → ✅ resolved — shared `FORMATION_SYSTEM_ACTOR_USERNAME` constant used in both places ([`e5db828`](https://github.com/linuxfoundation/lfx-self-serve/commit/e5db82869988eda5b312146153481a0ce8c73050))
- Missing direct tests → ⚠️ partially addressed — 4 tests added, but one is broken (see blocking finding below) and the precedence test's own target edge case isn't covered

**Bot reconciliation:** Cursor Bugbot's "Display-name spec skips checklist mock" finding on this commit is the same root cause as the blocking finding below (missing `proxyRequest` mock) — agreed, not a duplicate concern, just independent confirmation. Copilot's new finding ("Update comments to document cached actor display name") is also valid and in scope: the cache PII-inventory comments in `formation.service.ts`, `formation-people.constants.ts`, and `formation.interface.ts` now understate what's cached now that `name` is enriched — please fold that doc fix into the same pass as the test fix.

🔴 Blocking: 1 issue: New precedence test fails — missing checklist mock, `yarn test` is red on this head
🟡 Minor: 1 issue: Lone-part fallback branch (no top-level `name`) still untested
⚪ Nit: 1 issue: `ownerEnrichmentPromise` lacks a rejection handler until it's awaited ~60 lines later

🔴 **Needs changes before approval**

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/lfx-one/src/server/services/formation.service.spec.ts [hunk 11]
```diff
@@ -790,6 +791,48 @@ describe('FormationService', () => {
...
+
+      await expect(FormationService.userMetadataCacheValueForTests('u1')).resolves.toMatchObject({ name: 'Ada Lovelace' });
+      await expect(FormationService.userMetadataCacheValueForTests('u2')).resolves.toMatchObject({ name: 'Full Name' });
+    });
```
[the remark this answers is not in the document]
Comment (andrest50) [reply apps/lfx-one/src/server/services/formation.service.spec.ts:845]: Fixed — added `proxyRequest.mockResolvedValue(checklist([]))` and a proper `getProjectSettings` stub so `getFormationPeople` can complete the checklist gate before the metadata lookups. Also extended the test with a lone-part case (`{ given_name: 'Ada' }` → `'Ada'`) and an empty-profile case (`{}` → `null`) (commit 38838c9).

File: packages/shared/src/interfaces/formation-people.interface.ts [hunk 12]
```diff
@@ -42,11 +42,14 @@ export interface FormationPerson {
...
 /**
- * The three user-metadata fields the people card consumes — and the ONLY profile data the BFF
+ * The user-metadata fields the formation feature consumes — and the ONLY profile data the BFF
  * memoises across requests. The auth-service profile also carries address, phone and other PII
  * this feature never renders, so it is projected to this shape before any value enters the cache.
+ * `name` is derived from `given_name`/`family_name`/`name` and used to personalise item-owner
+ * and activity-actor displays (`FormationUser.name`).
  */
 export interface FormationPersonMetadata {
+  name: string | null;
```
[the remark this answers is not in the document]
Comment (andrest50) [reply packages/shared/src/interfaces/formation-people.interface.ts:52]: Fixed — updated the cache comment in `formation.service.ts` (line 134) and the TTL comment in `formation-people.constants.ts` (line 68) to list `name` alongside title/organization/picture (commit 38838c9).

File: apps/lfx-one/src/server/services/formation.service.spec.ts (hunk 8 above)
Comment (andrest50) [reply apps/lfx-one/src/server/services/formation.service.spec.ts:839]: Fixed — the test now stubs `proxyRequest` and `getProjectSettings` the same way all sibling `getFormationPeople` tests do. `yarn test` is green on this head (commit 38838c9).

File: apps/lfx-one/src/server/services/formation.service.ts (hunk 9 above)
Comment (andrest50) [reply apps/lfx-one/src/server/services/formation.service.ts:1119]: Fixed — the precedence test now includes a `u3` case with only `given_name` and no top-level `name`, asserting the resolved value is `'Ada'` (the lone part), plus a `u4` empty-profile case asserting `null` (commit 38838c9).

File: apps/lfx-one/src/server/services/formation.service.ts (hunk 10 above)
Comment (andrest50) [reply apps/lfx-one/src/server/services/formation.service.ts:169]: Fixed — appended `.catch(() => new Map<string, string>())` directly to the `enrichFormationUserRefs` call at creation so any rejection is absorbed immediately, with the same empty-map fallback that the `await` site would apply anyway (commit 38838c9).

Comment (dealako) [inline apps/lfx-one/src/server/services/formation.service.ts:169]: [minor] New `.catch()` on `ownerEnrichmentPromise` has no rationale comment

Issue: the added `.catch(() => new Map<string, string>())` here has no explanatory comment, while every other defensive `.catch()` in this same method is documented inline (e.g. the block comment a few lines up explicitly calls out the settings-read's `.catch()` below it and what it degrades to).
Proof: lines 169-172 add the `.catch()` with no accompanying comment.
Why it matters: the file's established convention documents every fail-safe/degrade path inline; an undocumented new one breaks that consistency and leaves a future reader to guess whether swallowing the rejection here is intentional resilience (it is — items keep their username-as-name placeholder) or an oversight.
Fix: add a short inline comment noting this falls back to an empty map (owners keep the username placeholder) if enrichment rejects outside `settleInBatches`' own per-item handling.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by dealako — APPROVED** [review]
Follow-up review of the latest commit: all prior review threads are resolved with verified fixes, tests pass (137/137), and no blocking or security/privacy issues remain. Two minor documentation-accuracy notes below.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by dealako** [comment]
Hey @andrest50 — reviewed the latest commit (`38838c9e`).

This follow-up cleanly closes out the prior round: the `.catch()` fallback on `ownerEnrichmentPromise` fixes the test regression, the two new fixture users (`u3`/`u4`) properly cover the `resolveDisplayName` edge cases (lone name part, empty profile), and the doc comments in `formation.service.ts` and `formation-people.constants.ts` were updated to mention the new `name` field. All 137 tests pass locally, format/lint are clean on the touched files, and there's no merge conflict against `main`.

👏 Nice work getting this across the line — the concurrency, precedence-logic, and test fixes all check out.

Prior feedback:
- ✅ Concurrency/serial-latency concern (owner enrichment) — resolved
- ✅ Name-fallback precedence logic (lone part / empty profile) — resolved
- ✅ Missing test coverage for display-name enrichment — resolved
- ✅ Format check / type-check failures — resolved
- ✅ Stale doc comments (partial: 2 of 3 sites) — see minor note below
- ✅ Test regression in this latest commit — resolved

🟡 Minor: 2 issues:
1. The new `.catch(() => new Map<string, string>())` on `ownerEnrichmentPromise` (formation.service.ts:169) lacks the inline rationale comment its sibling `.catch()` in the same method has — worth a one-liner so a future reader knows this is a deliberate best-effort degrade, not an oversight.
2. `packages/shared/src/interfaces/formation.interface.ts` (~line 369) still has a stale comment claiming `name` always mirrors the username for activity actors, which is no longer accurate now that display names are resolved. This wasn't touched by this PR's diff so I couldn't leave an inline comment on it, but Copilot's latest pass flagged the same thing independently — worth a quick follow-up.

✅ **Approved with minor comments**

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_016tYoyYBLo1hxj83QQodzX1

---
_Generated by [Claude Code](https://claude.ai/code)_
